eva-accession-clustering/src/main/java/uk/ac/ebi/eva/accession/clustering/batch/io/ClusteringWriter.java

```diff
@@ -165,185 +161,233 @@ private void getOrCreateClusteredVariantAccessions(List<? extends SubmittedVaria
+            return true;
+        }
 
-    private Set<SubmittedVariantEntity> getSubmittedVariantsRetainingRS(Map<Long,List<SubmittedVariantEntity>> submittedVariantsRSSplit, String assembly){
-        Set<SubmittedVariantEntity> submittedVariantRetainingRS = new HashSet<>();
-        for(Map.Entry<Long,List<SubmittedVariantEntity>> entry:submittedVariantsRSSplit.entrySet()){
-            try {
-                boolean isThereAnEntryForRSAndAssembly = clusteredService.getAllByAccession(entry.getKey()).stream()
-                        .map(accessionWrapper -> accessionWrapper.getData().getAssemblyAccession())
-                        .anyMatch(assemblyAccession -> assemblyAccession.equals(assembly));
+        return false;
+    }
 
-                if(!isThereAnEntryForRSAndAssembly){
-                    submittedVariantRetainingRS.add(entry.getValue().get(0));
-                }
-            }catch (AccessionMergedException| AccessionDoesNotExistException| AccessionDeprecatedException ex){
-                logger.error("exception occurred while getting variants with rs ID {" + entry.getKey() + "}. " + ex);
+    private void updateAllExistingHashesGroupByRS(Map<Long, Set<String>> allExistingHashesGroupByRS, String assembly,
+                                                  Long variantAccession) {
+        if (!allExistingHashesGroupByRS.containsKey(variantAccession)) {
+            Set<String> allHashForRS = getAllHashesForAssemblyAndRS(assembly, variantAccession);
+            if(allHashForRS.isEmpty()) {
+                allExistingHashesGroupByRS.put(variantAccession, new HashSet<>());
+            }else{
+                allExistingHashesGroupByRS.put(variantAccession, allHashForRS);
             }
         }
-        return submittedVariantRetainingRS;
     }
 
-    private List<GetOrCreateAccessionWrapper<IClusteredVariant, String, Long>> generateNewRSIdAndGetAccessionWrappers(
-            Collection<SubmittedVariantEntity> submittedVariantNewRS) throws AccessionCouldNotBeGeneratedException{
-        List<ClusteredVariant> clusteredVariantNewRS = submittedVariantNewRS.stream()
-                .map(this::toClusteredVariant)
-                .collect(Collectors.toList());
-        List<GetOrCreateAccessionWrapper<IClusteredVariant, String, Long>> accessionWrappers =
-                clusteredService.getOrCreate(clusteredVariantNewRS);
-        long newAccessions = accessionWrappers.stream().filter(GetOrCreateAccessionWrapper::isNewAccession).count();
-        clusteringCounts.addClusteredVariantsCreated(newAccessions);
+    private void insertAllEntriesInDB(List<ClusteredVariantEntity> clusteredVariantEntities,
+                                      Map<String, SubmittedVariantOperationEntity> mergeSVOE,
+                                      Map<Long, SubmittedVariantOperationEntity> rsSplitSVOE) {
 
-        return accessionWrappers;
-    }
+        mongoTemplate.insert(clusteredVariantEntities, ClusteredVariantEntity.class);
```

Comment:
What happens if the accessions are spread between dbSNP (<3B) and EVA (>=3B) ranges?

Reply:
At the moment we are not reading from the `dbsnpSubmittedVariantEntity` collection and it was assumed we would not be writing to the `dbsnpClusteredVariantEntity` collection but for this release we probably should be reading the remapped variants from the dbsnp collection. They will all be already clustered but we will need to add documents to the clustered collection with the new assembly and check for merge and split cases.

As part of the clustering we are only going to generate RS ids that are in the EVA ranges (>=3B)

Reply:
changed the logic to make entry into dbsnpClusteredVariantEntity or ClusteredVariantEntity based on accession.